bazel/test/python_test_repo/BUILD

```diff
@@ -60,3 +60,54 @@ py_test(
     ],
     python_version = "PY3",
 )
+
+# Test compatibility of py_proto_library and py_grpc_library rules with
+# proto_library targets as deps when the latter use import_prefix and/or
+# strip_import_prefix arguments
+proto_library(
+    name = "helloworld_moved_proto",
+    srcs = ["helloworld.proto"],
+    deps = [
+        "@com_google_protobuf//:duration_proto",
+        "@com_google_protobuf//:timestamp_proto",
+    ],
+    import_prefix = "google/cloud",
+    strip_import_prefix = ""
+)
+
+py_proto_library(
+    name = "helloworld_moved_py_pb2",
+    deps = [":helloworld_moved_proto"],
+)
+
+py_grpc_library(
+    name = "helloworld_moved_py_pb2_grpc",
+    srcs = [":helloworld_moved_proto"],
+    deps = [":helloworld_moved_py_pb2"],
+)
+
+py_test(
+    name = "import_moved_test",
+    main = "helloworld.py",
+    srcs = ["helloworld.py"],
+    deps = [
+        ":helloworld_moved_py_pb2",
+        ":helloworld_moved_py_pb2_grpc",
+        ":duration_py_pb2",
+        ":timestamp_py_pb2",
+    ],
+    imports = [
+        "_virtual_imports/helloworld_moved_proto",
```

Comment:
I somewhat wonder if it's considered good practice to depend on the `_virtual_imports` name. Is this part of Bazel's API?  Or is it considered an implementation detail? Who would be able to answer this question for us? @lberki perhaps?

Reply:
I think it is strictly a bad practice... Enabling the TODOs will allow us to avoid doing this, but with cost of making `py_proto_library` inconsistent with `cc_proto_library`.

Another option can be adding this iports inside `native.py_library` call under `py_proto_library`.

Reply:
If `cc_proto_library` already requires users to reference `_virtual_imports` in their `#include`s, that sounds as good as an API commitment from the Bazel team, but I would like to hear back from them before we make a decision.

Reply:
I don't know if _virtual_imports becomes visible for people or no (I would guess no). I know that this is where cc_proto_library puts its generated files, but users of the lib probably import it from different location (cc_proto_library is a native rule so it can do weird things, also C++ is compilable, so sources are just sources, and not used directly, compiled binary is used instead, which is different for python (where source and library are same thing)).

Reply:
Adding our Python expert, @brandjon for a more authoritative answer.

I'd much rather no one encodes `_virtual_imports` in BUILD files. That won't change (for a while), but it's considered an implementation detail and it's at Bazel's discretion whether it creates such a directory for any given `proto_library` rule.

The "official" way to do `LANG_proto_library` is to make them not macros, but rules (i.e. not `def bf_proto_library()` but `bf_proto_library = rule(...)`.) Then they can create language-specific providers (`PyInfo` in this case) and are able to do things like adding a new entry to the import path and are able to figure out he proto source root (by calling `dep[ProtoInfo].proto_source_root`). That way, dependent Python rules don't need to know how exactly they work.

Reply:
@lberki 

tl;dr; Is it possible to create PyInfo and or ProtoInfo provider instance from a Starlark rule? Last time I tried to create ProtoInfo from starlark I could not figure out how to do it and concluded that it could be created only from navite rules (but it was long time ago and I may be wrong here).

`py_proto_library` here consists of essentially two calls only: 1) generate python stubs, 2) call native py_library rule to make it a "library" (compile, if python was static languge). 

So it is as simple wrapper macro as it can be, and probably it is done this way because it allows to output a legit `py_library` without need to re-implement `py_library` here. 

`py_library` is a native rule with opaque implementation, so it is hard to figure out how to make py_proto_library in a form of a rule such that it can be consumed as a dependency (by `py_library`, `py_binary` and `py_test`).

Reply:
I have update the generation rule to return `PyInfo` provider as well with imports property set to point to `_virtual_imports`, and then pass that as deps to the subsequent py_library rule. This effectively allowed to hide `_virtual_imports` from the surface.